web/src/test/java/ua/com/vertex/logic/CourseLogicImplTest.java

```diff
@@ -35,6 +37,17 @@ public void setUp() {
     }
 
     @Test(expected = DataIntegrityViolationException.class)
+    public void getCoursesPerPageVerifyCourseDaoAndReturnException() throws Exception {
+        DataNavigator dataNavigator = new DataNavigator();
+        User teacher = new User.Builder().setUserId(7).getInstance();
+
+        when(courseDaoInf.getCoursesPerPage(dataNavigator, teacher))
+                .thenThrow(new DataIntegrityViolationException("Test"));
+        courseLogic.getCoursesPerPage(dataNavigator, teacher);
+        verify(courseDaoInf, times(1)).getCoursesPerPage(dataNavigator, teacher);
```

Comment:
if you expect exception, then add fail() just after the method call that should throw an exception. And I don't understand what verify is doing here in this case